generic3g/OuterMetaComponent.F90

```diff
@@ -277,6 +280,16 @@ module recursive subroutine recurse_(this, phase_idx, rc)
          integer, optional, intent(out) :: rc
       end subroutine recurse_
    
+      module recursive subroutine recurse_read_restart_(this, rc)
```

Comment:
I think the prefix `recurse_` is a bit overkill here.   Almost every procedure in the module should be renamed if this is the style you prefer. (In which case we'll discuss the merits either way.)

Reply:
Ignore that comment.  I see now where the name comes from.

The question we should consider is whether we want/need a different "recurse" procedure for each ESMF "run" method, or if we should just generalize the existing one.  I.e., add arguments to the interface for the ESMF_METHOD_FLAG and phase.

Reply:
Yes, we need a way to generalize the `recurse` procedure. The existing one would only call `initialize` recursively. That can be a separate issue and PR.

Reply:
OK - but do not forget ...